### PR TITLE
fix: Organization User Events' Spam

### PR DESCRIPTION
**Pull request by hariombalhara**
## What does this PR do?

Fixes organization-level spam detection for user events by deriving the organization ID from the hostname instead of relying solely on team hierarchy or user profile data. This ensures spam checks use the correct organization context based on the domain/subdomain being accessed.

### 1. Hostname-based Organization Resolution
- **Before**: Spam checks used `team?.parentId ?? eventType.profile?.organizationId` to determine the organization
- **After**: First attempts to extract org slug from hostname and fetch the organization ID, falling back to the original logic if unavailable

### 2. Implementation Details

**In `handleNewBooking.ts`:**
- Added hostname-based org resolution using `getOrgSlug()` and `OrganizationRepository.findBySlug()`
- Dynamically imports org-related modules to avoid circular dependencies
- Maintains backward compatibility by falling back to team/profile-based resolution
- Organization ID is now correctly passed to `spamCheckService.startCheck()`

**In `getEventTypesFromDB.ts`:**
- Updated to use `withSelectedCalendars()` for hosts and users
- Improved organization team event detection using `eventType.profile?.organizationId`

**In `defaultEvents.ts`:**
- Added `profile: null` to commons object for consistency

**In `orgDomains.ts`:**
- Added defensive checks to handle edge cases in org slug extraction

### 3. Comprehensive Test Coverage
Added integration tests in `spam-booking.integration-test.ts`:
- ✅ Verifies organization watchlist blocks bookings at org level
- ✅ Ensures personal user events are NOT blocked by org-level watchlist
- ✅ Tests fail-open behavior when SpamCheckService encounters errors

## Why This Change?

In multi-tenant deployments where organizations use custom domains/subdomains:
- Users can access the same event type via different organization domains
- Spam detection must use the organization context of the domain being accessed
- Previous logic could use the wrong organization for spam checks, allowing spam to bypass org-level blocks

## Example Scenario

**Before:**
```
User books acme-org-event.cal.com/sales/demo
→ Spam check uses eventType.profile?.organizationId (might be null or wrong org)
→ Org-level spam blocks don't work correctly
```

**After:**
```
User books acme-org-event.cal.com/sales/demo
→ Extract "acme-org-event" from hostname
→ Fetch organization ID from slug
→ Spam check uses correct org context
→ Org-level spam blocks work as expected
```

## Testing

- [x] Added integration tests for organization-level spam blocking
- [x] Added tests for personal event exemption from org blocks
- [x] Added fail-open error handling tests
- [x] Verified backward compatibility when hostname is not available
- [x] Lint and type checks pass

## Related Issues

Fixes organization user event spam detection issues where domain-based spam blocks were not being applied correctly.

### CODE REVIEW

File: packages/features/bookings/lib/handleNewBooking/test/spam-booking.integration-test.ts [hunk 1]
```diff
@@ -785,5 +785,94 @@ describe("handleNewBooking - Spam Detection", () => {
...
+              email: booker.email,
+              name: booker.name,
+              location: { optionValue: "", value: BookingLocations.CalVideo },
+            },
+          },
+        });
+
+        const createdBooking = await handleNewBooking({
+          bookingData: mockBookingData,
+          forcedSlug: org.slug,
```
Comment (hariombalhara) [inline packages/features/bookings/lib/handleNewBooking/test/spam-booking.integration-test.ts]: Using forcedSlug approach to tell the slug instead of hostname as it is not easily possible to mock ALLOWED_HOSTNAMES right now

**Review by alishaz-polymath — DISMISSED** [review]

File: packages/features/profile/repositories/ProfileRepository.ts [hunk 2]
```diff
@@ -626,6 +626,15 @@ export class ProfileRepository {
       });
     return profiles;
   }
+  static async findFirstForUserId({userId}: {userId: number }) {
+    const profiles = await prisma.profile.findMany({
+      where: {
+        userId: userId,
+      },
+      take: 1,
+    });
```
Comment (alishaz-polymath) [inline packages/features/profile/repositories/ProfileRepository.ts]: Why aren't we doing findFirst here instead of doing findMany with take 1?

File: packages/features/profile/repositories/ProfileRepository.ts [hunk 3]
```diff
@@ -626,6 +626,15 @@ export class ProfileRepository {
...
+    return profiles[0] ?? null;
```
Comment (hariombalhara) [inline packages/features/profile/repositories/ProfileRepository.ts]:
```suggestion
   return prisma.profile.findFirst({
    where: {
      userId: userId,
    },
  });
```

**Review by alishaz-polymath — DISMISSED** [review]

File: packages/features/ee/organizations/lib/orgDomains.ts [hunk 4]
```diff
@@ -15,8 +15,8 @@ const log = logger.getSubLogger({
  */
 export function getOrgSlug(hostname: string, forcedSlug?: string) {
   if (forcedSlug) {
-    if (process.env.NEXT_PUBLIC_IS_E2E) {
-      log.debug("Using provided forcedSlug in E2E", {
+    if (process.env.NEXT_PUBLIC_IS_E2E || process.env.INTEGRATION_TEST_MODE) {
+      log.debug("Using provided forcedSlug in E2E/Integration Test mode", {
```
Comment (hariombalhara) [inline packages/features/ee/organizations/lib/orgDomains.ts:19]: I think it is better to keep it still(not needed right now though) because it could be useful while integration testing in future

**Review by alishaz-polymath — APPROVED** [review]

File: packages/features/bookings/lib/handleNewBooking.ts [hunk 5]
```diff
@@ -427,6 +427,45 @@ export interface IBookingServiceDependencies {
...
+    // TODO: Moving it to instance based access through DI in a followup
+    const profile = await ProfileRepository.findFirstForUserId({
+      userId: eventType.userId,
+    });
+    eventOrganizationId = profile?.organizationId ?? null;
+    return eventOrganizationId;
+  }
+
+  return eventOrganizationId;
+}
```
[the remark this answers is not in the document]
Comment (hariombalhara) [reply packages/features/bookings/lib/handleNewBooking.ts:467]: Doesn't matter right now as we have just one profile per user.

File: packages/features/profile/repositories/ProfileRepository.ts [hunk 6]
```diff
@@ -626,6 +626,15 @@ export class ProfileRepository {
...
+  }
```
[the remark this answers is not in the document]
Comment (hariombalhara) [reply packages/features/profile/repositories/ProfileRepository.ts]: orderBy not required for now as we don't have multiple profiles in our system

**Review by emrysal — APPROVED** [review]
